providers/implementations/keymgmt/mlx_kmgmt.c

```diff
@@ -734,16 +734,22 @@ static void *mlx_kem_dup(const void *vkey, int selection)
         || (ret = OPENSSL_memdup(key, sizeof(*ret))) == NULL)
         return NULL;
 
-    if ((selection & OSSL_KEYMGMT_SELECT_KEYPAIR) == 0) {
+    switch (selection & OSSL_KEYMGMT_SELECT_KEYPAIR) {
+    case 0:
         ret->xkey = ret->mkey = NULL;
         return ret;
+    case OSSL_KEYMGMT_SELECT_KEYPAIR:
+        ret->mkey = EVP_PKEY_dup(key->mkey);
+        ret->xkey = EVP_PKEY_dup(key->xkey);
+        if (ret->xkey != NULL && ret->mkey != NULL)
+            return ret;
+        break;
+    default:
+        ERR_raise_data(ERR_LIB_PROV, PROV_R_UNSUPPORTED_SELECTION,
+                       "duplication of partial key material not supported");
```

Comment:
If `selection` is `OSSL_KEYMGMT_SELECT_PUBLIC_KEY` and the keys only contains public keys, then we will end up in this default case....but it should be fine to duplicate this right? The case we're worried about is if the keys contain private keys and the selection is `OSSL_KEYMGMT_SELECT_PUBLIC_KEY`.

Reply:
Yes, it is in principle fine to duplicate only the public key, if that's all we have, but it sets up unrealistic expectations that selective duplication is supported.  Since there is currently not code path to do that, I was conservative in only supporting what I know I can consistently deliver, which is to duplicate the verbatim key (or none of the key material).

Reply:
I can live with that.

Reply:
As we never use "public key only dup" it is fine as is IMO.